### PR TITLE
Feat(75): Current reserve

### LINKED ISSUE #75: [Landing] Implement current reserve section
## Description 

This issue is to implement the current reserve section.

This will make use of the existing data sources referenced below, please try and not modify the business logic as much as possible.

## Related info

- [Design desktop](https://www.figma.com/design/eW63rQE38ql8NGBYcFnCSJ/Analytics-portal?node-id=6855-3119&m=dev)
- [Design mobile](https://www.figma.com/design/eW63rQE38ql8NGBYcFnCSJ/Analytics-portal?node-id=6848-1192&m=dev)
- [Data source](https://github.com/mento-protocol/reserve-site/blob/master/src/components/StableTokens.tsx)

### PR DESCRIPTION
**Pull request by RyRy79261**
## Description 

This PR adds the current reserve view. Additionally, the component for the composition has been created as a reference for the next ticket.

## Other Changes

It also updated the loading state of the app; the `Amount` component and ratio section now have appropriate loading skeletons. 

see [this](https://reserve-site-git-feat-75-current-reserve-mentolabs.vercel.app/) vs [this](https://reserve-site-3el7nxuo7-mentolabs.vercel.app/)

## Related issues

- closes #75

### CODE REVIEW

File: src/hooks/useHoldings.ts [hunk 1]
```diff
@@ -29,10 +30,21 @@ const INITAL_DATA: HoldingsApi = {
     initalOtherToken,
     { ...initalOtherToken, token: "ETH" },
     { ...initalOtherToken, token: "DAI" },
+    { ...initalOtherToken, token: "USDC" },
```
Comment (RyRy79261) [inline src/hooks/useHoldings.ts:33]: Typo:
```suggestion
    { ...initialOtherToken, token: "USDC" },
```

File: src/pages/index.tsx [hunk 2]
```diff
@@ -60,15 +60,14 @@ export default function Home(props: Props) {
               stable assets.
             </p>
           </section>
-          <CardBackground>
-            <StableTokens />
-          </CardBackground>
-          <CardBackground className="mt-[56px]">
+          <div className="flex flex-col gap-14">
```
Comment (RyRy79261) [inline src/pages/index.tsx:63]: Why not just use the CardBackground as the div in this case?

Comment (denviljclarke) [reply src/pages/index.tsx:63]: We use it, but just internally now. The card background isn't displayed on mobile, so we can consider the card background a part of this component and then display it conditionally from within; this main page will only lay out the components now.

Comment (RyRy79261) [reply src/pages/index.tsx:63]: But I'm saying its redundant to have this div on line 63 when any of its styling could be applied to a card background item

Comment (denviljclarke) [reply src/pages/index.tsx:63]: That div is the wrapping div which separates the 3 cards, there's no card background where this div should be, it's just blank and we need some spacing

Comment (denviljclarke) [reply src/pages/index.tsx:63]: The card background which is still around stables will need to be moved as well as it's the same case where Stables doesn't need the card on mobile

Comment (RyRy79261) [reply src/pages/index.tsx:63]: Ahh didn't see the other children, sweet

**Review by denviljclarke — APPROVED** [review]
Ryan opened the PR and got started. However, I made the bulk of the changes. @RyRy79261 has reviewed and approved, but can't add an approval so I'm adding this approval on his behalf

File: src/components/Holdings.tsx [hunk 3]
```diff
@@ -82,54 +24,80 @@ export default function Holdings() {
...
+    <span
+      className={cn(
+        "inline-flex h-[58px] w-full items-center justify-between gap-5 rounded-md border-[1px] border-black bg-mento-mint p-4 font-fg text-[22px] md:h-[58px] md:w-[530px] md:gap-[90px] md:text-[26px]",
+      )}
+    >
+      <span>Total reserve holdings:</span>
+      {isLoading || !totalReserveValue ? (
+        <Skeleton className="h-[18px] w-[148px] bg-[#d8e9d0] md:h-[26px] md:w-[10.3rem]" />
+      ) : (
+        <span className="font-medium">{`${totalReserveValue.toFixed(2)} $`}</span>
```
Comment (chapati23) [inline src/components/Holdings.tsx:99]: 1. let's add comma separators here same as for all other $ amounts
2. Also, the $ is suffixed here, whereas in all subcomponents it's prefixed. Should use the same approach everywhere, probably prefer prefixed

File: src/service/rates.ts [hunk 4]
```diff
@@ -6,19 +6,19 @@ import { getOrSave, Cachable } from "src/service/cache";
...
+// import { getCMC02Price } from "src/providers/UbeSwapGraph";
 import getCoinMarketCapPrice from "src/providers/CoinMarketCap";
 import { Tokens } from "./Data";
 import { allOkOrThrow } from "src/utils/Result";
 
-async function fetchCMCO2price(): Promise<DuelResult> {
-  const cmco2 = await duel(getCMC02Price(), getCoinMarketCapPrice("MCO2"));
-  return cmco2;
-}
+// async function fetchCMCO2price(): Promise<DuelResult> {
```
Comment (chapati23) [inline src/service/rates.ts:14]: always best to leave a `// TODO` comment on outcommented stuff so tooling can pick it up and we won't forget it later

**Review by bayological — CHANGES_REQUESTED** [review]
The site looks good but this particular component needs to be changed slightly:
image

Even though it's displayed like that on the figma it doesn't make sense displaying the dollar sign at the end. Also we should format the number properly. 

There is already a function available to format this number, I added this in an [earlier commit](https://github.com/mento-protocol/reserve-site/pull/88/commits/ebdda826bf6445700a28933efcf845e6b5191dda) but somehow it was overwritten.

File: src/hooks/useHoldings.ts [hunk 5]
```diff
@@ -21,18 +22,29 @@ const initalOtherToken = {
...
 const INITAL_DATA: HoldingsApi = {
   celo: {
-    custody: initalToken,
-    unfrozen: initalToken,
-    frozen: initalToken,
+    custody: initialToken,
+    unfrozen: initialToken,
+    frozen: initialToken,
   },
   otherAssets: [
```
Comment (chapati23) [inline src/hooks/useHoldings.ts:29]: i know this wasn't strictly defined in the design but IMO the ordering seems quite random at the moment

- _either_ we should order by $ amount descending (i.e. largest holdings first, smallest last)
- _or_ at least group a bit more logically manually (i.e. USDT after sDAI feels off, intuitively would have expected from most to least used assets i.e. USDC => USDT => DAI => sDAI => euro stuff)

**Review by chapati23 — CHANGES_REQUESTED** [review]

File: src/service/rates.ts (hunk 4 above)
Comment (bayological) [reply src/service/rates.ts:14]: Jumping in here, this should just be removed. We no longer have cm02 in the reserve.

Comment (denviljclarke) [reply src/service/rates.ts:14]: Just this, or all reference to cmco2?

Comment (bayological) [reply src/service/rates.ts:14]: > Just this, or all reference to cmco2?

All references, but actually this was already done and merged to main [here](https://github.com/mento-protocol/reserve-site/pull/91)

Comment (denviljclarke) [reply src/service/rates.ts:14]: That's not all references. There are several more, including this commented-out code, which is still on main

Comment (bayological) [reply src/service/rates.ts]: > That's not all references. There are several more, including this commented-out code, which is still on main

My bad, I assumed all were removed. I'm thinking that actually it doesn't even matter now. We'll be redoing all of this logic from scratch so it won't be included.

**Review by bayological — APPROVED** [review]

**Review by chapati23 — APPROVED** [review]

**Pull request merged**